src/groups/group.jl

```diff
@@ -1061,3 +1060,8 @@ function group_exp!(G::MultiplicationGroup, q, X)
         "group_exp! not implemented on $(typeof(G)) for vector $(typeof(X)) and element $(typeof(q)).",
     )
 end
+
+function group_log!(G::MultiplicationGroup, X, q)
+    q isa AbstractMatrix && return log_safe!(X, q)
```

Comment:
Why don't you put `::AbstractMatrix` in method signature? I think we generally want to get rid of "method not implemented" errors.

Reply:
If we define `group_log!(G::MultiplicationGroup, X, q::AbstractMatrix)` then any time we want to overload `group_log!` for a single group `MyGroup` under the multiplication operation, we then have to overload `group_log!(G::MyGroup, X, q::AbstractMatrix)` and `group_log!(G::MyGroup, X, q)` to not have ambiguity. At the moment it's not a problem, because only `SpecialOrthogonal` has a custom overload, but it could become a nuisance in the future. What do you think?

Reply:
I went ahead and changed it. We can always switch it back in the future.

Reply:
I think this makes – for now – the interface/function signature nice.
My remark on your point would be: Would a `MyGroup` not be `<:AbstractGroup` and hence be more like a sister type of `MultiplicationGroup` such that we would not have the ambiguity problem? I am not 100% sure, but we can surely improve the signature later, if need be.

Reply:
I think it's quite unlikely we would ever want to define `group_log!(G::MyGroup, X, q)` with unrestricted `q`, and in any case we can adjust this method later.